rust/main/agents/scraper/src/store/deliveries.rs

```diff
@@ -41,3 +51,30 @@ impl HyperlaneLogStore<Delivery> for HyperlaneDbStore {
         Ok(stored as u32)
     }
 }
+
+#[async_trait]
+impl HyperlaneSequenceAwareIndexerStoreReader<Delivery> for HyperlaneDbStore {
+    /// Gets a delivered message by its nonce.
+    async fn retrieve_by_sequence(&self, sequence: u32) -> Result<Option<Delivery>> {
+        let delivery = self
+            .db
+            .retrieve_delivered_message_by_sequence(
+                self.domain.id(),
+                &self.mailbox_address,
+                sequence,
+            )
+            .await?;
+        Ok(delivery)
+    }
+
+    /// Gets the block number at which the log occurred.
+    async fn retrieve_log_block_number_by_sequence(&self, sequence: u32) -> Result<Option<u64>> {
```

Comment:
nit: any thoughts / intuition on the runtime cost of this requiring 3 round trips? wondering if it's worth a single query that joins